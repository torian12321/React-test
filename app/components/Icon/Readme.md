With [ReactIcons](https://www.npmjs.com/package/react-icons) the icons are converted into SVG, so there is no need to add external .css or .js libraries.

Check available icons from [the list](https://gorangajic.github.io/react-icons/index.html)

Basic:
```
<Icon />
```

Custom Icon:
```
<div>
	<Icon name='twitter' />
	<Icon name='facebook' />
	<Icon name='cog' />
</div>
```