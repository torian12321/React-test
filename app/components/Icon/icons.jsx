/**
* Social
*/

export const twitter = [
  { d: 'M2 4 C6 8 10 12 15 11 A6 6 0 0 1 22 4 A6 6 0 0 1 26 6 A8 8 0 0 0 31 4 A8 8 0 0 1 28 8 A8 8 0 0 0 32 7 A8 8 0 0 1 28 11 A18 18 0 0 1 10 30 A18 18 0 0 1 0 27 A12 12 0 0 0 8 24 A8 8 0 0 1 3 20 A8 8 0 0 0 6 19.5 A8 8 0 0 1 0 12 A8 8 0 0 0 3 13 A8 8 0 0 1 2 4' }
]

export const facebook = [
  { d: 'M8 12 L13 12 L13 8 C13 2 17 1 24 2 L24 7 C20 7 19 7 19 10 L19 12 L24 12 L23 18 L19 18 L19 30 L13 30 L13 18 L8 18 z' }
]

export const google = [
  { d: 'M16.319 13.713v5.487h9.075c-0.369 2.356-2.744 6.9-9.075 6.9-5.463 0-9.919-4.525-9.919-10.1s4.456-10.1 9.919-10.1c3.106 0 5.188 1.325 6.375 2.469l4.344-4.181c-2.788-2.612-6.4-4.188-10.719-4.188-8.844 0-16 7.156-16 16s7.156 16 16 16c9.231 0 15.363-6.494 15.363-15.631 0-1.050-0.113-1.85-0.25-2.65l-15.113-0.006z' }
]

export const instagram = [
  { d: 'M16 2.881c4.275 0 4.781 0.019 6.462 0.094 1.563 0.069 2.406 0.331 2.969 0.55 0.744 0.288 1.281 0.638 1.837 1.194 0.563 0.563 0.906 1.094 1.2 1.838 0.219 0.563 0.481 1.412 0.55 2.969 0.075 1.688 0.094 2.194 0.094 6.463s-0.019 4.781-0.094 6.463c-0.069 1.563-0.331 2.406-0.55 2.969-0.288 0.744-0.637 1.281-1.194 1.837-0.563 0.563-1.094 0.906-1.837 1.2-0.563 0.219-1.413 0.481-2.969 0.55-1.688 0.075-2.194 0.094-6.463 0.094s-4.781-0.019-6.463-0.094c-1.563-0.069-2.406-0.331-2.969-0.55-0.744-0.288-1.281-0.637-1.838-1.194-0.563-0.563-0.906-1.094-1.2-1.837-0.219-0.563-0.481-1.413-0.55-2.969-0.075-1.688-0.094-2.194-0.094-6.463s0.019-4.781 0.094-6.463c0.069-1.563 0.331-2.406 0.55-2.969 0.288-0.744 0.638-1.281 1.194-1.838 0.563-0.563 1.094-0.906 1.838-1.2 0.563-0.219 1.412-0.481 2.969-0.55 1.681-0.075 2.188-0.094 6.463-0.094zM16 0c-4.344 0-4.887 0.019-6.594 0.094-1.7 0.075-2.869 0.35-3.881 0.744-1.056 0.412-1.95 0.956-2.837 1.85-0.894 0.888-1.438 1.781-1.85 2.831-0.394 1.019-0.669 2.181-0.744 3.881-0.075 1.713-0.094 2.256-0.094 6.6s0.019 4.887 0.094 6.594c0.075 1.7 0.35 2.869 0.744 3.881 0.413 1.056 0.956 1.95 1.85 2.837 0.887 0.887 1.781 1.438 2.831 1.844 1.019 0.394 2.181 0.669 3.881 0.744 1.706 0.075 2.25 0.094 6.594 0.094s4.888-0.019 6.594-0.094c1.7-0.075 2.869-0.35 3.881-0.744 1.050-0.406 1.944-0.956 2.831-1.844s1.438-1.781 1.844-2.831c0.394-1.019 0.669-2.181 0.744-3.881 0.075-1.706 0.094-2.25 0.094-6.594s-0.019-4.887-0.094-6.594c-0.075-1.7-0.35-2.869-0.744-3.881-0.394-1.063-0.938-1.956-1.831-2.844-0.887-0.887-1.781-1.438-2.831-1.844-1.019-0.394-2.181-0.669-3.881-0.744-1.712-0.081-2.256-0.1-6.6-0.1v0z' },
  { d: 'M16 7.781c-4.537 0-8.219 3.681-8.219 8.219s3.681 8.219 8.219 8.219 8.219-3.681 8.219-8.219c0-4.537-3.681-8.219-8.219-8.219zM16 21.331c-2.944 0-5.331-2.387-5.331-5.331s2.387-5.331 5.331-5.331c2.944 0 5.331 2.387 5.331 5.331s-2.387 5.331-5.331 5.331z' },
  { d: 'M26.462 7.456c0 1.060-0.859 1.919-1.919 1.919s-1.919-0.859-1.919-1.919c0-1.060 0.859-1.919 1.919-1.919s1.919 0.859 1.919 1.919z' }
]

export const youtube = [
  { d: 'M31.681 9.6c0 0-0.313-2.206-1.275-3.175-1.219-1.275-2.581-1.281-3.206-1.356-4.475-0.325-11.194-0.325-11.194-0.325h-0.012c0 0-6.719 0-11.194 0.325-0.625 0.075-1.987 0.081-3.206 1.356-0.963 0.969-1.269 3.175-1.269 3.175s-0.319 2.588-0.319 5.181v2.425c0 2.587 0.319 5.181 0.319 5.181s0.313 2.206 1.269 3.175c1.219 1.275 2.819 1.231 3.531 1.369 2.563 0.244 10.881 0.319 10.881 0.319s6.725-0.012 11.2-0.331c0.625-0.075 1.988-0.081 3.206-1.356 0.962-0.969 1.275-3.175 1.275-3.175s0.319-2.587 0.319-5.181v-2.425c-0.006-2.588-0.325-5.181-0.325-5.181zM12.694 20.15v-8.994l8.644 4.513-8.644 4.481z' }
]

export const linkedin = [
  { d: 'M11.2,16.7104 C11.2,14.2384 11.1184,12.1728 11.0368,10.3888 L16.7872,10.3888 L17.072,13.1456 L17.1936,13.1456 C18.0032,11.848 19.9872,9.9456 23.3088,9.9456 C27.36,9.9456 30.4,12.6576 30.4,18.4928 L30.4,30.4 L24,30.4 L24,19.4224 C24,16.872 23.0288,14.9264 20.8,14.9264 C19.1008,14.9264 18.2496,16.3024 17.8032,17.4368 C17.64,17.8432 17.6,18.4096 17.6,18.9776 L17.6,30.4 L11.2,30.4 L11.2,16.7104 Z M8,4.8 C8,6.4992 6.864,7.9616 4.7984,7.9616 C2.8544,7.9616 1.6,6.5824 1.6,4.8832 C1.6,3.1392 2.816,1.6 4.8,1.6 C6.784,1.6 7.9616,3.056 8,4.8 Z M1.6,30.4 L1.6,9.6 L8,9.6 L8,30.4 L1.6,30.4 Z' }
]

/**
* Other
*/

export const calendar = [
  { d: 'M2 4 L6 4 L6 2 A2 2 0 0 1 10 2 L10 4 L22 4 L22 2 A2 2 0 0 1 26 2 L26 4 L30 4 L30 10 L2 10 M2 12 L30 12 L30 30 L2 30' }
]

export const chat = [
  { d: 'M32 16 A16 12 0 0 0 0 16 A16 12 0 0 0 16 28 L18 28 C20 30 24 32 28 32 C27 31 26 28 26 25.375 L26 25.375 A16 12 0 0 0 32 16' }
]

export const check = [
  { d: 'M1 14 L5 10 L13 18 L27 4 L31 8 L13 26 z' }
]

export const chevron = [
  { d: 'M12 1 L26 16 L12 31 L8 27 L18 16 L8 5 z' }
]

export const close = [
  { d: 'M4 8 L8 4 L16 12 L24 4 L28 8 L20 16 L28 24 L24 28 L16 20 L8 28 L4 24 L12 16 z' }
]

export const cog = [
  { d: 'M14 0 H18 L19 6 L20.707 6.707 L26 3.293 L28.707 6 L25.293 11.293 L26 13 L32 14 V18 L26 19 L25.293 20.707 L28.707 26 L26 28.707 L20.707 25.293 L19 26 L18 32 L14 32 L13 26 L11.293 25.293 L6 28.707 L3.293 26 L6.707 20.707 L6 19 L0 18 L0 14 L6 13 L6.707 11.293 L3.293 6 L6 3.293 L11.293 6.707 L13 6 L14 0 z M16 10 A6 6 0 0 0 16 22 A6 6 0 0 0 16 10' }
]

export const dollar = [
  { d: 'M17.6 26.808v3.592h-3.2v-3.429c-2.739-0.16-4.906-0.942-6.786-2.875l2.749-2.784c1.374 1.392 3.237 1.856 5.251 1.856 2.504 0 3.848-0.958 3.848-2.723 0-0.773-0.213-1.422-0.672-1.856-0.427-0.402-0.915-0.619-1.923-0.773l-2.627-0.371c-1.862-0.278-3.235-0.866-4.214-1.826-1.037-1.051-1.557-2.474-1.557-4.331 0-3.448 2.211-5.989 5.931-6.56v-3.128h3.2v3.091c2.211 0.232 3.944 0.992 5.464 2.482l-2.686 2.691c-1.374-1.331-3.022-1.515-4.459-1.515-2.259 0-3.358 1.267-3.358 2.784 0 0.557 0.184 1.146 0.642 1.578 0.427 0.403 1.13 0.742 2.016 0.866l2.563 0.371c1.986 0.278 3.237 0.835 4.154 1.701 1.162 1.114 1.68 2.723 1.68 4.672 0 3.6-2.507 5.859-6.014 6.488z' }
]

export const heart = [
  { d: 'M0 10 C0 6, 3 2, 8 2 C12 2, 15 5, 16 6 C17 5, 20 2, 24 2 C30 2, 32 6, 32 10 C32 18, 18 29, 16 30 C14 29, 0 18, 0 10' }
]

export const home = [
  { d: 'M16 0 L32 16 L28 16 L28 30 L20 30 L20 20 L12 20 L12 30 L4 30 L4 16 L0 16 Z' }
]

export const loading = [
  { d: 'M32,16.5322245 C31.9588157,14.3081247 31.4913515,12.0909439 30.6279637,10.0636341 C29.7674119,8.03452807 28.5184129,6.19642412 26.9851198,4.67752183 C25.4527291,3.15748857 23.6354642,1.95585863 21.6684149,1.16311019 C19.7025257,0.367301455 17.5887855,-0.0128399168 15.5005035,0.0332640333 C13.4119637,0.0759085239 11.3327049,0.544665281 9.43133132,1.40893139 C7.52847533,2.2703368 5.8046002,3.52019958 4.38048741,5.0538711 C2.9553434,6.58674428 1.82951057,8.40382536 1.08748439,10.3693971 C0.342622356,12.3340374 -0.0123101712,14.4449064 0.0322255791,16.5322245 C0.0736676737,18.6198753 0.514449144,20.6951518 1.32556697,22.593131 C2.1340423,24.4925073 3.30666667,26.2131892 4.74502316,27.6342287 C6.18260624,29.0563992 7.8861148,30.1790603 9.72754884,30.9180541 C11.5682095,31.6599751 13.5445398,32.0125073 15.5005035,31.966736 C17.456854,31.9238254 19.398574,31.4826112 21.1746546,30.6722994 C22.9518953,29.8645821 24.5620785,28.6937547 25.8913837,27.2579459 C27.22172,25.8229356 28.2710493,24.1232765 28.960999,22.2873015 C29.3810272,21.1743534 29.6660947,20.0120416 29.8163303,18.8345613 C29.8564834,18.8369563 29.8968298,18.838553 29.9375629,18.838553 C31.0766083,18.838553 32,17.8854719 32,16.7096549 C32,16.6499127 31.9969708,16.5908358 31.9923303,16.5322911 L32,16.5322911 L32,16.5322245 Z M28.722852,22.1853805 C27.9664532,23.9550936 26.8703968,25.5583534 25.5267835,26.8815967 C24.1839436,28.2058378 22.5942558,29.2495301 20.8784371,29.9349023 C19.1631339,30.622736 17.3242779,30.9478586 15.5005035,30.9022869 C13.6764068,30.8591102 11.8720322,30.4453721 10.2213092,29.6890146 C8.56974824,28.935052 7.0733857,27.8431933 5.83882377,26.5052474 C4.60335952,25.1680998 3.63046928,23.5858628 2.99259617,21.8796175 C2.35227392,20.1737048 2.05070695,18.3465114 2.09466264,16.5322245 C2.13662034,14.717605 2.52416516,12.9260042 3.2307432,11.2868191 C3.93512991,9.64690229 4.95474723,8.16099792 6.20342397,6.93555094 C7.45152064,5.70917256 8.92745217,4.74444906 10.5175911,4.11289813 C12.1075368,3.47888565 13.8089829,3.18130561 15.5005035,3.22661123 C17.1923464,3.27005405 18.8593112,3.65631601 20.3846767,4.35871933 C21.9106224,5.05892723 23.2932286,6.07188358 24.4329829,7.31196674 C25.5736395,8.55131809 26.4699617,10.0161331 27.0558228,11.5926486 C27.6438751,13.1690977 27.9188882,14.8544532 27.8751259,16.5322245 L27.8827956,16.5322245 C27.8781551,16.5907692 27.8751259,16.6498462 27.8751259,16.7095884 C27.8751259,17.8077006 28.6805076,18.7112183 29.7147553,18.8259792 C29.5141833,19.9842994 29.1820665,21.1151435 28.722852,22.1853805 L28.722852,22.1853805 Z' }
]

export const mail = [
  { d: 'M0 6 L16 16 L32 6 z M0 9 L0 26 L32 26 L32 9 L16 19 z' }
]

export const menu = [
  { d: 'M0,14.125 L0,17.875 C0,18.22 0.448,18.5 1,18.5 L31,18.5 C31.552,18.5 32,18.22 32,17.875 L32,14.125 C32,13.78 31.552,13.5 31,13.5 L1,13.5 C0.448,13.5 0,13.78 0,14.125 Z M0,2.625 L0,6.375 C0,6.72 0.448,7 1,7 L31,7 C31.552,7 32,6.72 32,6.375 L32,2.625 C32,2.28 31.552,2 31,2 L1,2 C0.448,2 0,2.28 0,2.625 Z M0,25.625 L0,29.375 C0,29.72 0.448,30 1,30 L31,30 C31.552,30 32,29.72 32,29.375 L32,25.625 C32,25.28 31.552,25 31,25 L1,25 C0.448,25 0,25.28 0,25.625 Z' }
]

export const minus = [
  { d: 'M0 13v6c0 0.552 0.448 1 1 1h30c0.552 0 1-0.448 1-1v-6c0-0.552-0.448-1-1-1h-30c-0.552 0-1 0.448-1 1z' }
]

export const pin = [
  { d: 'M4 12 A12 12 0 0 1 28 12 C28 20, 16 32, 16 32 C16 32, 4 20 4 12 M11 12 A5 5 0 0 0 21 12 A5 5 0 0 0 11 12 Z' }
]

export const plus = [
  { d: 'M31 12h-11v-11c0-0.552-0.448-1-1-1h-6c-0.552 0-1 0.448-1 1v11h-11c-0.552 0-1 0.448-1 1v6c0 0.552 0.448 1 1 1h11v11c0 0.552 0.448 1 1 1h6c0.552 0 1-0.448 1-1v-11h11c0.552 0 1-0.448 1-1v-6c0-0.552-0.448-1-1-1z' }
]

export const search = [
  { d: 'M31.0082809,27.2313682 L23.4284569,20.7845177 C22.6444747,20.079534 21.8064942,19.7555417 21.1295093,19.7865408 C22.9184676,17.6905895 23.9994436,14.9716526 23.9994436,11.9997216 C23.9994436,5.37287538 18.6265671,0 11.9997209,0 C5.37287467,0 -9.9475983e-14,5.37287538 -9.9475983e-14,11.9997216 C-9.9475983e-14,18.6265678 5.37287467,23.9994432 11.9997209,23.9994432 C14.9716533,23.9994432 17.6905902,22.9184683 19.7865404,21.1295097 C19.7555413,21.806494 20.0795342,22.6444745 20.7845173,23.4284564 L27.2313689,31.0082805 C28.3353422,32.2342521 30.1383004,32.3382496 31.2382756,31.2382752 C32.3382489,30.1383006 32.2352516,28.3353426 31.0082809,27.2313682 Z M12.0007218,20.000536 C7.58282489,20.000536 4.00090667,16.418619 4.00090667,12.0007216 C4.00090667,7.582824 7.58282489,4.0009072 12.0007218,4.0009072 C16.4186187,4.0009072 20.0005351,7.582824 20.0005351,12.0007216 C20.0005351,16.418619 16.4186187,20.000536 12.0007218,20.000536 Z' }
]

export const star = [
  { d: 'M16 0 L21 11 L32 12 L23 19 L26 31 L16 25 L6 31 L9 19 L0 12 L11 11' }
]

export const target = [
  { d: 'M16 1.333q2.99 0 5.703 1.161t4.677 3.125 3.125 4.677 1.161 5.703-1.161 5.703-3.125 4.677-4.677 3.125-5.703 1.161-5.703-1.161-4.677-3.125-3.125-4.677-1.161-5.703 1.161-5.703 3.125-4.677 4.677-3.125 5.703-1.161zM16 4q-2.438 0-4.661 0.953t-3.828 2.557-2.557 3.828-0.953 4.661 0.953 4.661 2.557 3.828 3.828 2.557 4.661 0.953 4.661-0.953 3.828-2.557 2.557-3.828 0.953-4.661-0.953-4.661-2.557-3.828-3.828-2.557-4.661-0.953zM16 6.667q1.896 0 3.625 0.74t2.979 1.99 1.99 2.979 0.74 3.625-0.74 3.625-1.99 2.979-2.979 1.99-3.625 0.74-3.625-0.74-2.979-1.99-1.99-2.979-0.74-3.625 0.74-3.625 1.99-2.979 2.979-1.99 3.625-0.74zM16 9.333q-1.354 0-2.589 0.531t-2.125 1.422-1.422 2.125-0.531 2.589 0.531 2.589 1.422 2.125 2.125 1.422 2.589 0.531 2.589-0.531 2.125-1.422 1.422-2.125 0.531-2.589-0.531-2.589-1.422-2.125-2.125-1.422-2.589-0.531zM16 12q1.656 0 2.828 1.172t1.172 2.828-1.172 2.828-2.828 1.172-2.828-1.172-1.172-2.828 1.172-2.828 2.828-1.172zM16 14.667q-0.552 0-0.943 0.391t-0.391 0.943 0.391 0.943 0.943 0.391 0.943-0.391 0.391-0.943-0.391-0.943-0.943-0.391z' }
]

export const warning = [
  { d: 'M15 0 H17 L32 29 L31 30 L1 30 L0 29 z M19 8 L13 8 L14 20 L18 20 z M16 22 A3 3 0 0 0 16 28 A3 3 0 0 0 16 22' }
]