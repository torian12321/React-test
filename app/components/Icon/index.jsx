var
React     = require('react'),
PropTypes = require('prop-types'),
classNames= require('classnames'),
//styles    = require('./styles'),
icons     = require('./icons');


const Icon = props => {
  const
  iconPaths = props.name ? icons[props.name] : props.paths,
  styles = {
      width : `${props.width}em`,
      height: props.height ? `${props.height}em` : null,
      fill  : props.color,
      ...props.styles
  };

  return (
    <svg
      viewBox  ={`0 0 ${props.viewBox} ${props.viewBox}`}
      style    ={styles}
      className={classNames(
        'icon',
        props.className
      )}
    >
      {iconPaths && iconPaths.map((pathProps, i) => <path {...pathProps} key={i} />)}
    </svg>
  )
}

Icon.propTypes = {
  name     : PropTypes.string,
  className: PropTypes.string,
  color    : PropTypes.string,
  rotate   : PropTypes.number,
  viewBox  : PropTypes.number,

  /** Size in ems */
  width: PropTypes.number,

  /** Size in ems */
  height: PropTypes.number,

  /** The paths of a custom icon */
  paths : PropTypes.array,

  /** Aditional Styles */
  styles : PropTypes.object
}

Icon.defaultProps = {
  name   : 'home',
  paths  : [],
  rotate : 0,
  width  : 1,
  viewBox: 32,
}

module.exports = Icon;