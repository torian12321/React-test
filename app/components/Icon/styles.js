export default (props, traits) => {
  const {
    colors
  } = traits

  const {
    color,
    rotate,
    size,
    spin,
    styles
  } = props

  const spinStyles = spin ? {
    animation: 'spin 1s linear infinite',
    '@keyframes spin': {
      '0%': {
        transform: 'rotate(0deg)'
      },
      '100%': {
        transform: 'rotate(360deg)'
      }
    }
  } : {}

  return {
    root: {
      fill: colors[color] || color,
      display: 'inline-block',
      width: `${size}em`,
      height: `${size}em`,
      transform: `rotate(${rotate}deg)`,
      ...spinStyles,
      ...styles
    }
  }
}