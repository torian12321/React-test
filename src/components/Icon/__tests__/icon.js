import React    from 'react';
import renderer from 'react-test-renderer';
import Icon     from '../index';

describe('component', () => {
  describe('<Icon />', () => {

    it('should render correctly', () => {
      expect(renderer.create(
        <Icon />
      )).toMatchSnapshot();
    });

    it('Should render a mail icon', () => {
      expect(renderer.create(
        <Icon name='mail' />
      )).toMatchSnapshot();
    });

    it('Should render with grey color', () => {
      expect(renderer.create(
        <Icon color='#DDD' />
      )).toMatchSnapshot();
    });

    it('Should render rotated', () => {
        expect(renderer.create(
          <Icon rotate={180} />
        )).toMatchSnapshot();
      });

  });
});
