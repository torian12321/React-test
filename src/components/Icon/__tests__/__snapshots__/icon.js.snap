// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Icon /> Should render a mail icon 1`] = `
<svg
  className="icon"
  style={
    Object {
      "fill": null,
      "height": null,
      "transform": null,
      "width": "1em",
    }
  }
  viewBox="0 0 32 32"
>
  <g>
    <path
      d="M0 6l16 10L32 6zm0 3v17h32V9L16 19z"
    />
  </g>
</svg>
`;

exports[`component <Icon /> Should render rotated 1`] = `
<svg
  className="icon"
  style={
    Object {
      "fill": null,
      "height": null,
      "transform": "rotate(180deg)",
      "width": "1em",
    }
  }
  viewBox="0 0 32 32"
>
  <g>
    <path
      d="M16 0l16 16h-4v14h-8V20h-8v10H4V16H0z"
    />
  </g>
</svg>
`;

exports[`component <Icon /> Should render with grey color 1`] = `
<svg
  className="icon"
  style={
    Object {
      "fill": "#DDD",
      "height": null,
      "transform": null,
      "width": "1em",
    }
  }
  viewBox="0 0 32 32"
>
  <g>
    <path
      d="M16 0l16 16h-4v14h-8V20h-8v10H4V16H0z"
    />
  </g>
</svg>
`;

exports[`component <Icon /> should render correctly 1`] = `
<svg
  className="icon"
  style={
    Object {
      "fill": null,
      "height": null,
      "transform": null,
      "width": "1em",
    }
  }
  viewBox="0 0 32 32"
>
  <g>
    <path
      d="M16 0l16 16h-4v14h-8V20h-8v10H4V16H0z"
    />
  </g>
</svg>
`;
